adapters/syslog/syslog.go

```diff
@@ -91,7 +115,9 @@ func (a *SyslogAdapter) Stream(logstream chan *router.Message) {
 		if err != nil {
 			log.Println("syslog:", err)
 			if reflect.TypeOf(a.conn).String() != "*net.UDPConn" {
-				return
+				if ok := a.maybeReconnect(err); !ok {
+					return
```

Comment:
@keis i don't know if there "should" be.  With UDP transport failed sends aren't retried either. Anyhow, i'd rather first hear back from @progrium whether he is intending to go forward with this.